evm_loader/program/src/evm/database.rs

```diff
@@ -76,6 +78,12 @@ pub trait Database {
         data: &[u8],
         is_static: bool,
     ) -> Option<Result<Vec<u8>>>;
+
+    #[cfg(not(target_os = "solana"))]
+    fn collect_log(&mut self, log: Log);
+
+    #[cfg(not(target_os = "solana"))]
+    fn logs(&self) -> Vec<Log>;
```

Comment:
I wouldn't say I like the idea of using conditional compilation widely. Why you cannot add your own `Log` structure (and don't use it from `web3::types::Log`) and special method `fn log(&self, log: Log)` into `Database`/`Backend` and then implement it different ways for NeonEVM program and emulation logic?
`ExecutorState` and `SyncedExecutorState`, in this case, should just pass it to the backend logc.